Reviewing docs for errors and mistakes uptill concepts section (#53)
openmrs/openmrs-contrib-rest-api-docs: source/includes/Concepts/concept.md

* The concept is the basic element of flexibility in OpenMRS. Concepts are the individual data points collected from a 
population of patients. Concepts include both questions and answers. 

* For example, blood type data is collected for a patient.  The question is, "What is the blood type for the patient?", with a
set of discrete answers of "A, B, AB or O."  We use concepts in OpenMRS to implement this. The question is a concept 
("blood type") and each response ("A," "B," "AB" and "O") is also a concept.  For this one question, a total of 5 concepts are required.

* What about a question where the answer is not a discrete answer?  If the question is "What is the name of your first pet?",
the answer would be expressed in a text box.  It would not be possible to provide a complete list of every possible name for
your pet.  In this example, there would be one concept -- "name of first pet."

> * The bottom line is that if you need a medical word within your electronic records system, it needs to be defined within the concept dictionary, more details about all the possible concepts will be given in a later section.

Reviewer: ```suggestion
* The bottom-line is that if you need a medical word within your electronic records system, it needs to be defined within the concept dictionary, more details about all the possible concepts will be given in a later section.
```

    *name* | `String` | ConceptNames are the words or phrases used to express the idea of a Concept within a particular locale
    *term* | `String` | Medical coding term or OpenMRS concept dictionary term that could be mapped to a concept
    *source* | `String` | A concept can have any number of mappings to any number of other vocabularies. Other vocabularies are called "concept sources" in OpenMRS (ie. LOINC, SNOMED, ICD-9, ICD10, RxNORM, etc), but the concept source can also be a custom (ie. org.openmrs.module.mdrtb, PIH, AMPATH, MVP, etc.). Every concept can define a string for its mapping in any "concept source" defined in the database

>     *class* | `String` | The concept's class provides a useful way to narrow down the scope of the information that the concept is intended to capture.  In this way, the class is helpful for data extraction.  This classification elaborates how a concept will be represented (i.e., as a question or an answer) when the information is stored.  OpenMRS contains several default classes to use when defining concepts, but implementation sites may also create additional custom concept classes.

Reviewer: ```suggestion
    *class* | `String` | The concept's class provides a useful way to narrow down the scope of the information that the concept is intended to capture.  In this way, the class is helpful for data extraction.  This classification elaborates how a concept will be represented (i.e. as a question or an answer) when the information is stored.  OpenMRS contains several default classes to use when defining concepts, but implementation sites may also create additional custom concept classes.
```

Reviewer 2: @sanjulamadurapperuma actually I think its notationally correct to use a comma after i.e. Like the one I have used above If its not I will change it 🙂

Reviewer: It's actually in American English. They use **i.e,** with the comma instead of the last full-stop/period. If we are okay with following that notation, then I am +1 to have **i.e,** instead of **i.e.,** which seems out-of-place.

Reviewer 2: Okay @sanjulamadurapperuma I will make the changes 🙂

    
* ### Query concept by UUID.

    Retrieve a concept by its UUID. Returns a `404 Not Found` status if concepts not exist. If the user is not logged in to perform this action, a `401 Unauthorized` status returned.
    
```console
GET /concept/:target_concept_uuid
```
   
## Create a concept

> * To Create a concept, you need to specify below attributes in the request body. If you are not logged in to perform this action,

Reviewer: ```suggestion
* To create a concept, you need to specify below attributes in the request body. If you are not logged in to perform this action,
```

Reviewer 2: @sanjulamadurapperuma all the operations were starting from a capital letter so should I change them all like update delete as well ?

Reviewer: I think it is better to because it actually is a verb, not the operation name. @dkayiwa @Akayeshmantha appreciate your thoughts on this.

## Update a concept

*  Update a target concept with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 

> status if concept not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status returned.

Reviewer: ```suggestion
status if concept not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status is returned.
```

* ### List all concept mappings for a concept.

    Retrieve all **concept mapping** subresources of a **concept** resource by target_concept_uuid. Returns a 
    `404 Not Found` status if concept mapping not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status returned.

```console
GET /concept/:target_concept_uuid/mapping 
```

* ### List concept mapping by its UUID and parent concept UUID.
    
     Retrieve a **concept mapping** subresources of a **concept** resource. Returns a 
     `404 Not Found` status if concept mapping not exists. If you are not logged in to perform this action, a `401 Unauthorized` status returned.
     
```console
GET /concept/:target_concept_uuid/mapping/:target_concept_mapping_uuid
```
## Create a concept mapping with properties

> * To Create a concept mapping subresource for a specific concept resource, you need to specify below attributes in the request body.

Reviewer: ```suggestion
* To create a concept mapping sub-resource for a specific concept resource, you need to specify below attributes in the request body.
```

Author: Okay @sanjulamadurapperuma I will make the changes 🙂

 
## Update a concept mapping

> * Updates a concept mapping subresource value with given UUID, this method will only modify the value of the subresource. Returns a `404 Not Found` status if concept mapping not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status

Reviewer: ```suggestion
* Updates a concept mapping sub-resource value with given UUID, this method will only modify the value of the sub-resource. Returns a `404 Not Found` status if concept mapping not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status
```

```
## Delete a concept mapping

> * Delete or Voided a target concept mapping subresource by its UUID. Returns a `404 Not Found` status if concept mapping not exists. 

Reviewer: ```suggestion
* Delete or Voided a target concept mapping sub-resource by its UUID. Returns a `404 Not Found` status if concept mapping not exists. 
```

* ### List all concept names for a concept.

>     Retrieve all **concept name** subresources of a **concept** resource by target_concept_uuid. Returns a 

Reviewer: ```suggestion
    Retrieve all **concept name** sub-resources of a **concept** resource by target_concept_uuid. Returns a 
```

>     `404 Not Found` status if concept name not exists. If the user isnot logged in to perform this action, a `401 Unauthorized` status returned.

Reviewer: ```suggestion
    `404 Not Found` status if concept name not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status returned.
```

```console
GET /concept/:target_concept_uuid/name 
```

* ### List concept name it's UUID and parent concept UUID.
    
     Retrieve a **concept name** subresources of a **concept** resource. Returns a 
     `404 Not Found` status if concept name not exists. If you are not logged in to perform this action, a `401 Unauthorized` status returned.
     
```console
GET /concept/:target_concept_uuid/name/:target_concept_name_uuid
```
## Create a concept name with properties

> * To Create a concept name subresource for a specific concept resource, you need to specify below attributes in the request body.

Reviewer: ```suggestion
* To Create a concept name sub-resource for a specific concept resource, you need to specify below attributes in the request body.
```

 
## Update concept name

> * Updates a concept name subresource value with given UUID, this method will only modify value of the subresource. Returns a `404 Not Found` status if concept name not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status

Reviewer: ```suggestion
* Updates a concept name sub-resource value with given UUID, this method will only modify value of the sub-resource. Returns a `404 Not Found` status if concept name not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status
```

```
## Delete concept name

> * Delete or retire a target concept name subresource by its UUID. Returns a `404 Not Found` status if concept name not exists. 

Reviewer: ```suggestion
* Delete or retire a target concept name sub-resource by its UUID. Returns a `404 Not Found` status if concept name not exists. 
```

* ### List all concept attributes for a concept.

    Retrieve all **concept attribute** subresources of a **concept** resource by target_concept_uuid. Returns a 
    `404 Not Found` status if a concept attribute not exists. If the user isnot logged in to perform this action, a `401 Unauthorized` status returned.

```console
GET /concept/:target_concept_uuid/attribute 
```

* ### List concept attribute by its UUID and parent concept UUID.
    

>      Retrieve a **concept attribute** subresources of a **concept** resource. Returns a 

Reviewer: ```suggestion
     Retrieve a **concept attribute** sub-resources of a **concept** resource. Returns a 
```

     `404 Not Found` status if a concept attribute not exists. If you are not logged in to perform this action, a `401 Unauthorized` status returned.
     
```console
GET /concept/:target_concept_uuid/attribute/:target_concept_attribute_uuid
```
## Create a concept attribute with properties

> * To Create a concept attribute subresource for a specific concept resource, you need to specify below attributes in the request body.

Reviewer: ```suggestion
* To create a concept attribute sub-resource for a specific concept resource, you need to specify below attributes in the request body.
```

```
## Delete concept attribute

> * Delete or retire a target concept attribute subresource by its UUID. Returns a `404 Not Found` status if concept attribute not exists. 

Reviewer: ```suggestion
* Delete or retire a target concept attribute sub-resource by its UUID. Returns a `404 Not Found` status if concept attribute not exists. 
```

* ### List all concept descriptions for a concept.

    Retrieve all **concept description** subresources of a **concept** resource by target_concept_uuid. Returns a 
    `404 Not Found` status if concept description not exists. If the user is not logged in to perform this action, a `401 Unauthorized` status returned.

```console
    GET /concept/:target_concept_uuid/description 
```

* ### List concept description by its UUID and parent concept UUID.
    
     Retrieve a **concept description** subresources of a **concept** resource. Returns a 
     `404 Not Found` status if concept description not exists. If you are not logged in to perform this action, a `401 Unauthorized` status returned.
     
```console
GET /concept/:target_concept_uuid/description/:target_concept_description_uuid
```
## Create a concept description with properties

> * To Create a concept description subresource for a specific concept resource you need to specify below attributes in the request body.

Reviewer: ```suggestion
* To create a concept description sub-resource for a specific concept resource you need to specify below attributes in the request body.
```

```console
POST concept/:target_concept_uuid/description
{
  "description": "Pregnancy terminated by spontaneous abortion.",
  "locale": "en"
}
```
## Delete concept description

> * Delete or retire a target concept description subresource by its UUID. Returns a `404 Not Found` status if concept description not exists. 

Reviewer: ```suggestion
* Delete or retire a target concept description sub-resource by its UUID. Returns a `404 Not Found` status if concept description not exists. 
```
